src/key/key.js

```diff
@@ -384,7 +420,8 @@ Key.prototype.decrypt = async function(passphrases, keyId = null) {
     await Promise.all(passphrases.map(async function(passphrase) {
       try {
         await key.keyPacket.decrypt(passphrase);
-        decrypted = true;
+        // If we are decrypting a single key packet, we also validate it
+        decrypted = keyId ? await key.keyPacket.validate() : true;
```

Comment:
if `decrypted` is false, then an error is thrown at line 430 - however, it won't be "Invalid key" error, since I never set it specifically, so I will change that 
However, I do agree that throwing on all `validate` is probably best, since we do not want this to go unnoticed in any case..

Reply:
Hmhm, it's a bit pointless indeed. Maybe we can remove this in v5